lib/Doctrine/DBAL/Connection.php

```diff
@@ -968,13 +968,14 @@ public function executeCacheQuery($query, $params, $types, QueryCacheProfile $qc
      * @param Closure $function The transformation function that is applied on each row.
      *                           The function receives a single parameter, an array, that
      *                           represents a row of the result set.
+     * @param mixed[] $types    The types the previous parameters are in.
      *
      * @return mixed[] The projected result of the query.
      */
-    public function project($query, array $params, Closure $function)
+    public function project($query, array $params, Closure $function, array $types = [])
```

Comment:
In my understanding a new optional parameter at the end of an existing function doesn't produce a BC or am I wrong?

Reply:
It produces a BC for extending classes, see https://3v4l.org/nlEKK . You can get the new parameter thanks to `func_get_args` without declaring it, though. It's very ugly, but it will work and it will be BC.

Reply:
@greg0ire thanks for your explanation. I changed the PR description to BC.